In [1]:
library(tidyverse)
nyc<- read_csv("flights.csv")
airport <- read_csv("airport.csv")
airline <- read_csv("airline.csv")    

Warning message in system("timedatectl", intern = TRUE):
“running command 'timedatectl' had status 141”
Warning message:
“Failed to locate timezone database”
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.5     ✔ purrr   0.3.4
✔ tibble  3.1.5     ✔ dplyr   1.0.7
✔ tidyr   1.1.4     ✔ stringr 1.4.0
✔ readr   2.0.2     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter()  masks stats::filter()
✖ purrr::flatten() masks jsonlite::flatten()
✖ dplyr::lag()     masks stats::lag()

New names:
* `` -> ...1

Rows: 336776 Columns: 20

── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (4): carrier, tailnum, origin, dest
dbl  (15): ...1, year, month, day, dep_time, sched_dep_time, dep_delay, arr_...
dttm  (1): time_hour


ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_t

In [2]:
nyc %>% glimpse()
glimpse(airport)
glimpse(airline)

Rows: 336,776
Columns: 20
$ ...1           <dbl> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, …
$ year           <dbl> 2013, 2013, 2013, 2013, 2013, 2013, 2013, 2013, 2013, 2…
$ month          <dbl> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…
$ day            <dbl> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…
$ dep_time       <dbl> 517, 533, 542, 544, 554, 554, 555, 557, 557, 558, 558, …
$ sched_dep_time <dbl> 515, 529, 540, 545, 600, 558, 600, 600, 600, 600, 600, …
$ dep_delay      <dbl> 2, 4, 2, -1, -6, -4, -5, -3, -3, -2, -2, -2, -2, -2, -1…
$ arr_time       <dbl> 830, 850, 923, 1004, 812, 740, 913, 709, 838, 753, 849,…
$ sched_arr_time <dbl> 819, 830, 850, 1022, 837, 728, 854, 723, 846, 745, 851,…
$ arr_delay      <dbl> 11, 20, 33, -18, -25, 12, 19, -14, -8, 8, -2, -3, 7, -1…
$ carrier        <chr> "UA", "UA", "AA", "B6", "DL", "UA", "B6", "EV", "B6", "…
$ flight         <dbl> 1545, 1714, 1141, 725, 461, 1696, 507, 5708, 79, 301, 4…
$ tailnum     

How many distance in 2013 and which the most distance flight

In **August** is most distance in 2013 **(50150427.74)**
 , **Flight 51** is Longest distance **(4983 Mile)**
 and **Flight 461** is Lowest distance **(80 mile)**

In [3]:
nyc %>%
  select(month,distance) %>%
  group_by(month) %>%
  summarise(Km. = sum(distance)*1.61)
    

A tibble: 12 × 2 
 
 month Km. 
 <dbl> <dbl> 
 
 
 1 43773976 
 2 40210569 
 3 46979214 
 4 47377943 
 5 48258346 
 6 48068785 
 7 50150210 
 8 50150428 
 9 46225396 
 10 48319458 
 11 46109946 
 12 48226075

In [4]:
nyc %>%
    group_by(flight) %>%
    summarise(distance = max(distance)) %>%
    arrange(desc(distance)) %>%
    head(5)
    
    

A tibble: 5 × 2 
 
 flight distance 
 <dbl> <dbl> 
 
 
 51 4983 
 15 4963 
 887 3370 
 11 2586 
 23 2586

In [5]:
nyc %>%
    group_by(flight) %>%
    summarise(distance = max(distance,rm.na = T)) %>%
    arrange(desc(distance)) %>%
    tail(5)

A tibble: 5 × 2 
 
 flight distance 
 <dbl> <dbl> 
 
 
 3864 94 
 3884 94 
 3986 94 
 4013 94 
 4616 80

# Which carrier in **May** is fly the most
**United Air Lines Inc. 4960 **

In [6]:
nyc %>% 
    filter(month == 5) %>%
    count(carrier) %>%
    arrange(desc(n)) %>%
    left_join(airline,by ="carrier") %>%
    head(5)

A tibble: 5 × 3 
 
 carrier n name 
 <chr> <int> <chr> 
 
 
 UA 4960 United Air Lines Inc. 
 EV 4817 ExpressJet Airlines Inc. 
 B6 4576 JetBlue Airways 
 DL 4082 Delta Air Lines Inc. 
 AA 2803 American Airlines Inc.

# where the lowest destination
**La Guardia Airport and Blue Grass airport**

In [7]:
nyc %>%
    count(dest) %>%
    arrange(desc(n)) %>%
    left_join(airport,by = c("dest" = "faa")) %>%
    tail(5)

A tibble: 5 × 10 
 
 dest n ...1 name lat lon alt tz dst tzone 
 <chr> <int> <dbl> <chr> <dbl> <dbl> <dbl> <dbl> <chr> <chr> 
 
 
 MTJ 15 928 Montrose Regional Airport 38.50979 -107.89424 5759 -7 A America/Denver 
 SBN 10 1193 South Bend Rgnl 41.70866 -86.31725 799 -5 A America/New_York 
 ANC 8 129 Ted Stevens Anchorage Intl 61.17436 -149.99636 152 -9 A America/Anchorage 
 LEX 1 783 Blue Grass 38.03650 -84.60589 979 -5 A America/New_York 
 LGA 1 787 La Guardia 40.77725 -73.87261 22 -5 A America/New_York

In [8]:
airport <- subset(airport,select = -c(1))

# Which month has the most flights in Miami Intl?
#December

In [9]:
nyc %>%
    filter(dest == "MIA") %>%
    group_by(month) %>%
    count(dest) %>%
    arrange(desc(n)) %>%
    left_join(airport,by = c("dest" = "faa"))


A grouped_df: 12 × 10 
 
 month dest n name lat lon alt tz dst tzone 
 <dbl> <chr> <int> <chr> <dbl> <dbl> <dbl> <dbl> <chr> <chr> 
 
 
 12 MIA 1104 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 3 MIA 1017 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 5 MIA 991 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 1 MIA 981 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 7 MIA 978 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 8 MIA 975 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 11 MIA 972 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 4 MIA 971 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 10 MIA 971 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 6 MIA 954 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 9 MIA 912 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York 
 2 MIA 902 Miami Intl 25.79325 -80.29056 8 -5 A America/New_York

# JFK longest and lowest flights distance in 2013
#Longest 4983 mile 
#Lowest   94 mile

In [10]:
nyc %>%
    group_by(origin) %>%
    filter(origin == "JFK") %>%
    summarise(Longest_dist = max(distance),
              Lowest_dist  = min(distance))

A tibble: 1 × 3 
 
 origin Longest_dist Lowest_dist 
 <chr> <dbl> <dbl> 
 
 
 JFK 4983 94

# Which airline has the least delays in JFK
# HA : Hawaiian Airlines Inc.

In [11]:
nyc %>%
    group_by(carrier)%>%
    filter(origin == "JFK") %>%
    summarize(avg_delay = mean(arr_delay, na.rm = T)) %>%
            arrange(avg_delay) %>%
            left_join(airline,by = "carrier")
    

A tibble: 10 × 3 
 
 carrier avg_delay name 
 <chr> <dbl> <chr> 
 
 
 HA -6.915205 Hawaiian Airlines Inc. 
 DL -2.379250 Delta Air Lines Inc. 
 AA 2.081250 American Airlines Inc. 
 US 2.114035 US Airways Inc. 
 UA 2.510496 United Air Lines Inc. 
 VX 2.827722 Virgin America 
 9E 8.843327 Endeavor Air Inc. 
 B6 8.893702 JetBlue Airways 
 MQ 12.468704 Envoy Air 
 EV 17.788839 ExpressJet Airlines Inc.

In [12]:
install.packages("RPostgreSQL")
library(RPostgreSQL)

Error in download.file(url, destfile, method, mode = "wb", ...) : 
  ignoring SIGPIPE signal


Warning message in download.packages(pkgs, destdir = tmpd, available = available, :
“download of package ‘RPostgreSQL’ failed”


ERROR: ERROR: Error in library(RPostgreSQL): there is no package called ‘RPostgreSQL’


In [ ]:
# connect data base
con <- dbConnect(
  PostgreSQL(),
  host = "tiny.db.elephantsql.com",
  dbname = "sxkiwtvf",
  port = 5432,
  user = "sxkiwtvf",
  password = "j-Hpowam7_AYVhDHDDC-BzfoknQjan4-"
)

In [ ]:
# create dataframe
fav_game <- data.frame(
    game_num = 1:5,
    game_name = c("Counter-Strike: Global Offensive", "	Dota 2", "	Apex Legends", "Grand Theft Auto V", "Rust"),
    Dev = c("Valve", "Bizzard", "Respawn Entertainment" , "Rockstar", "Facepunch")
)

# create table in database
dbWriteTable(con, "fav_game", fav_game)


In [ ]:
fav_place <- data.frame(
    id = 1:5
    country = c("China", "Korea", "Laos", "Zombia", "Belgium" ),
    region = c("Asia", "Asia", "Asia", "Africa", "Europe")

) 
    
    # create table in database
dbWriteTable(con, "fav_place", fav_place)

In [ ]:
movie_list <- data.frame(
  id = 1:5,
  movie_name = c("Lost", "The Midnight Guy", "Top Gun: Maverick", "Knives Out", "La La Land"),
  score = c(3, 2, 5, 3, 4)
  year = c("2010", "2016", "2022", "2017", "2016")
)

dbWriteTable(con, "books_list", books_list)

In [ ]:
# list tables in database
dbListTables(con)

#Query
dbGetQuery(con, "SELECT * FROM fav_place")

In [ ]:
# disconnect database
dbDisconnect(con)